### ISSUE TITLE
digital_ocean backup_enabled: yes is enabling private_networking

### ISSUE THREAD

**Issue by regentgal** [ansible/ansible-modules-core#3557]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

digital_ocean
##### ANSIBLE VERSION

```
$ ansible --version
ansible 2.0.2.0
  config file = /Users/jln/git/jpw/black-ops/ansible.cfg
  configured module search path = Default w/o overrides
```
##### SUMMARY

Attempting to create a DO droplet with private_networking and back_up enabled. If I set only private_networking: yes, private networking is not enabled on the droplet. If I set only backups_enabled: yes, backups are not enabled, but private networking is. If I set both private_networking: yes and backups_enabled: yes, private networking is enabled but backups are not.
##### STEPS TO REPRODUCE

```
  - name: create database droplet
    digital_ocean:
      state: present
      command: droplet
      name: testWithNetworking
      size_id: 2gb
      region_id: nyc3
      image_id: ubuntu-14-04-x64
      unique_name: yes
      private_networking: yes
    register: droplet

  - name: create database droplet
    digital_ocean:
      state: present
      command: droplet
      name: testWithBackups
      size_id: 2gb
      region_id: nyc3
      image_id: ubuntu-14-04-x64
      unique_name: yes
      backups_enabled: yes
    register: droplet

  - name: create database droplet
    digital_ocean:
      state: present
      command: droplet
      name: testWithBoth
      size_id: 2gb
      region_id: nyc3
      image_id: ubuntu-14-04-x64
      unique_name: yes
      private_networking: yes
      backups_enabled: yes
    register: droplet
```

##### EXPECTED RESULTS

I expected three droplets:
- One with Private Networking enabled, but not Backups.
- One with Backups enabled, but not Private Networking.
- One with Private Networking and Backups enabled.
##### ACTUAL RESULTS

![image](https://cloud.githubusercontent.com/assets/3778233/14833019/10217c7a-0bcb-11e6-874b-32e88241d04c.png)

![image](https://cloud.githubusercontent.com/assets/3778233/14833008/fcfb4888-0bca-11e6-8a8c-0c3039486766.png)

![image](https://cloud.githubusercontent.com/assets/3778233/14833033/26cb467c-0bcb-11e6-9eb1-4c54b8714e4b.png)

``````
Using /Users/jln/git/jpw/black-ops/ansible.cfg as config file
Loaded callback default of type stdout, v2.0

PLAYBOOK: create_db_server.yml *************************************************
2 plays in create_db_server.yml

PLAY [local] *******************************************************************

TASK [setup] *******************************************************************
ESTABLISH LOCAL CONNECTION FOR USER: jln
localhost EXEC /bin/sh -c '( umask 22 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1461701603.74-47521402779626 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1461701603.74-47521402779626 `" )'
localhost PUT /var/folders/6g/mhwwnt7j3259lddvql8jhkbw0000gn/T/tmpureuWb TO /Users/jln/.ansible/tmp/ansible-tmp-1461701603.74-47521402779626/setup
localhost EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 python /Users/jln/.ansible/tmp/ansible-tmp-1461701603.74-47521402779626/setup; rm -rf "/Users/jln/.ansible/tmp/ansible-tmp-1461701603.74-47521402779626/" > /dev/null 2>&1'
ok: [localhost]

TASK [droplet : create database droplet] ***************************************
task path: /Users/jln/git/jpw/black-ops/roles/droplet/tasks/main.yml:10
ESTABLISH LOCAL CONNECTION FOR USER: jln
localhost EXEC /bin/sh -c '( umask 22 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1461701604.63-264720731229174 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1461701604.63-264720731229174 `" )'
localhost PUT /var/folders/6g/mhwwnt7j3259lddvql8jhkbw0000gn/T/tmp1K1RbK TO /Users/jln/.ansible/tmp/ansible-tmp-1461701604.63-264720731229174/digital_ocean
localhost EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 python /Users/jln/.ansible/tmp/ansible-tmp-1461701604.63-264720731229174/digital_ocean; rm -rf "/Users/jln/.ansible/tmp/ansible-tmp-1461701604.63-264720731229174/" > /dev/null 2>&1'
changed: [localhost] => {"changed": true, "droplet": {"backup_ids": [], "created_at": "2016-04-26T20:13:25Z", "disk": 40, "drives": [], "features": ["ipv6", "virtio"], "id": 14185559, "image": {"created_at": "2016-02-23T23:01:40Z", "distribution": "Ubuntu", "id": 15943679, "min_disk_size": 20, "name": "14.04.4 x64", "public": true, "regions": ["nyc1", "sfo1", "nyc2", "ams2", "sgp1", "lon1", "nyc3", "ams3", "fra1", "tor1", "sfo2"], "size_gigabytes": 1.58, "slug": "ubuntu-14-04-x64", "type": "snapshot"}, "ip_address": "104.236.71.85", "kernel": null, "locked": false, "memory": 2048, "name": "testWithNetworking", "networks": {"v4": [{"gateway": "104.236.64.1", "ip_address": "104.236.71.85", "netmask": "255.255.192.0", "type": "public"}], "v6": [{"gateway": "2604:A880:0800:0010:0000:0000:0000:0001", "ip_address": "2604:A880:0800:0010:0000:0000:0B1E:1001", "netmask": 64, "type": "public"}]}, "next_backup_window": null, "region": {"available": true, "features": ["private_networking", "backups", "ipv6", "metadata"], "name": "New York 3", "sizes": ["32gb", "16gb", "2gb", "1gb", "4gb", "8gb", "512mb", "64gb", "48gb"], "slug": "nyc3"}, "size": {"available": true, "disk": 40, "memory": 2048, "price_hourly": 0.02976, "price_monthly": 20.0, "regions": ["ams1", "ams2", "ams3", "fra1", "lon1", "nyc1", "nyc2", "nyc3", "sfo1", "sgp1", "tor1"], "slug": "2gb", "transfer": 3.0, "vcpus": 2}, "size_slug": "2gb", "snapshot_ids": [], "status": "active", "tags": [], "vcpus": 2}, "invocation": {"module_args": {"api_token": null, "backups_enabled": false, "command": "droplet", "id": null, "image_id": "ubuntu-14-04-x64", "name": "testWithNetworking", "private_networking": true, "region_id": "nyc3", "size_id": "2gb", "ssh_key_ids": null, "ssh_pub_key": null, "state": "present", "unique_name": true, "user_data": null, "virtio": true, "wait": true, "wait_timeout": 300}, "module_name": "digital_ocean"}}

TASK [droplet : create database droplet] ***************************************
task path: /Users/jln/git/jpw/black-ops/roles/droplet/tasks/main.yml:22
ESTABLISH LOCAL CONNECTION FOR USER: jln
localhost EXEC /bin/sh -c '( umask 22 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1461701648.46-30799619553555 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1461701648.46-30799619553555 `" )'
localhost PUT /var/folders/6g/mhwwnt7j3259lddvql8jhkbw0000gn/T/tmp62vX08 TO /Users/jln/.ansible/tmp/ansible-tmp-1461701648.46-30799619553555/digital_ocean
localhost EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 python /Users/jln/.ansible/tmp/ansible-tmp-1461701648.46-30799619553555/digital_ocean; rm -rf "/Users/jln/.ansible/tmp/ansible-tmp-1461701648.46-30799619553555/" > /dev/null 2>&1'
changed: [localhost] => {"changed": true, "droplet": {"backup_ids": [], "created_at": "2016-04-26T20:14:09Z", "disk": 40, "drives": [], "features": ["private_networking", "virtio"], "id": 14185593, "image": {"created_at": "2016-02-23T23:01:40Z", "distribution": "Ubuntu", "id": 15943679, "min_disk_size": 20, "name": "14.04.4 x64", "public": true, "regions": ["nyc1", "sfo1", "nyc2", "ams2", "sgp1", "lon1", "nyc3", "ams3", "fra1", "tor1", "sfo2"], "size_gigabytes": 1.58, "slug": "ubuntu-14-04-x64", "type": "snapshot"}, "ip_address": "10.132.15.191", "kernel": null, "locked": false, "memory": 2048, "name": "testWithBackups", "networks": {"v4": [{"gateway": "10.132.0.1", "ip_address": "10.132.15.191", "netmask": "255.255.0.0", "type": "private"}, {"gateway": "104.236.192.1", "ip_address": "104.236.236.56", "netmask": "255.255.192.0", "type": "public"}], "v6": []}, "next_backup_window": null, "region": {"available": true, "features": ["private_networking", "backups", "ipv6", "metadata"], "name": "New York 3", "sizes": ["32gb", "16gb", "2gb", "1gb", "4gb", "8gb", "512mb", "64gb", "48gb"], "slug": "nyc3"}, "size": {"available": true, "disk": 40, "memory": 2048, "price_hourly": 0.02976, "price_monthly": 20.0, "regions": ["ams1", "ams2", "ams3", "fra1", "lon1", "nyc1", "nyc2", "nyc3", "sfo1", "sgp1", "tor1"], "slug": "2gb", "transfer": 3.0, "vcpus": 2}, "size_slug": "2gb", "snapshot_ids": [], "status": "active", "tags": [], "vcpus": 2}, "invocation": {"module_args": {"api_token": null, "backups_enabled": true, "command": "droplet", "id": null, "image_id": "ubuntu-14-04-x64", "name": "testWithBackups", "private_networking": false, "region_id": "nyc3", "size_id": "2gb", "ssh_key_ids": null, "ssh_pub_key": null, "state": "present", "unique_name": true, "user_data": null, "virtio": true, "wait": true, "wait_timeout": 300}, "module_name": "digital_ocean"}}

TASK [droplet : create database droplet] ***************************************
task path: /Users/jln/git/jpw/black-ops/roles/droplet/tasks/main.yml:34
ESTABLISH LOCAL CONNECTION FOR USER: jln
localhost EXEC /bin/sh -c '( umask 22 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1461701690.74-258804638768663 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1461701690.74-258804638768663 `" )'
localhost PUT /var/folders/6g/mhwwnt7j3259lddvql8jhkbw0000gn/T/tmppdM6x9 TO /Users/jln/.ansible/tmp/ansible-tmp-1461701690.74-258804638768663/digital_ocean
localhost EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 python /Users/jln/.ansible/tmp/ansible-tmp-1461701690.74-258804638768663/digital_ocean; rm -rf "/Users/jln/.ansible/tmp/ansible-tmp-1461701690.74-258804638768663/" > /dev/null 2>&1'
changed: [localhost] => {"changed": true, "droplet": {"backup_ids": [], "created_at": "2016-04-26T20:14:52Z", "disk": 40, "drives": [], "features": ["private_networking", "ipv6", "virtio"], "id": 14185606, "image": {"created_at": "2016-02-23T23:01:40Z", "distribution": "Ubuntu", "id": 15943679, "min_disk_size": 20, "name": "14.04.4 x64", "public": true, "regions": ["nyc1", "sfo1", "nyc2", "ams2", "sgp1", "lon1", "nyc3", "ams3", "fra1", "tor1", "sfo2"], "size_gigabytes": 1.58, "slug": "ubuntu-14-04-x64", "type": "snapshot"}, "ip_address": "10.132.2.33", "kernel": null, "locked": false, "memory": 2048, "name": "testWithBoth", "networks": {"v4": [{"gateway": "10.132.0.1", "ip_address": "10.132.2.33", "netmask": "255.255.0.0", "type": "private"}, {"gateway": "104.131.0.1", "ip_address": "104.131.52.215", "netmask": "255.255.192.0", "type": "public"}], "v6": [{"gateway": "2604:A880:0800:0010:0000:0000:0000:0001", "ip_address": "2604:A880:0800:0010:0000:0000:0B1E:2001", "netmask": 64, "type": "public"}]}, "next_backup_window": null, "region": {"available": true, "features": ["private_networking", "backups", "ipv6", "metadata"], "name": "New York 3", "sizes": ["32gb", "16gb", "2gb", "1gb", "4gb", "8gb", "512mb", "64gb", "48gb"], "slug": "nyc3"}, "size": {"available": true, "disk": 40, "memory": 2048, "price_hourly": 0.02976, "price_monthly": 20.0, "regions": ["ams1", "ams2", "ams3", "fra1", "lon1", "nyc1", "nyc2", "nyc3", "sfo1", "sgp1", "tor1"], "slug": "2gb", "transfer": 3.0, "vcpus": 2}, "size_slug": "2gb", "snapshot_ids": [], "status": "active", "tags": [], "vcpus": 2}, "invocation": {"module_args": {"api_token": null, "backups_enabled": true, "command": "droplet", "id": null, "image_id": "ubuntu-14-04-x64", "name": "testWithBoth", "private_networking": true, "region_id": "nyc3", "size_id": "2gb", "ssh_key_ids": null, "ssh_pub_key": null, "state": "present", "unique_name": true, "user_data": null, "virtio": true, "wait": true, "wait_timeout": 300}, "module_name": "digital_ocean"}}

TASK [droplet : debug] *********************************************************
task path: /Users/jln/git/jpw/black-ops/roles/droplet/tasks/main.yml:61
ok: [localhost] => {
    "msg": "testWithBoth(14185606) - 10.132.2.33"
}

PLAY [db] **********************************************************************
skipping: no hosts matched

PLAY RECAP *********************************************************************
localhost                  : ok=5    changed=3    unreachable=0    failed=0```
``````

**Comment by chouseknecht** [comment]
@zbal can you take a look at this?

**Comment by regentgal** [comment]
This is behavior is a result of dopy 3.5. The same behavior occurs using dopy directly to create the droplets with private_networking and backups_enabled.

Using dopy 3.7 directly works, but has know issues with Ansible: https://github.com/ansible/ansible-modules-core/issues/2509.

So, I guess I'll go try and figure out that one...

**Comment by ansibot** [comment]
@mgregson, @alukovenko, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by alukovenko** [comment]
It seems that dopy brings a lot of problems, so it would be better to get rid of using dopy at all (take a look at #4219) - wontfix
